pkg/integrationtests/integrationtests.go

```diff
@@ -0,0 +1,233 @@
+package integrationtests
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/r3labs/diff/v3"
+)
+
+type Issue struct {
+	Severity string `json:"severity"`
+	Title    string `json:"title"`
+	Detail   string `json:"detail"`
+	Name     string `json:"name"`
+}
+
+type JsonReport struct {
+	Id              string             `json:"id"`
+	Version         string             `json:"version"`
+	PluginValidator map[string][]Issue `json:"plugin-validator"`
+}
+
+var files = map[string]JsonReport{
```

Comment:
effectively the integration tests will run for each of the files defined in this map and compare that the validator outputs the same errors and warnings. It'll exit 1 if any of the keys changed.

Currently the reporting is not optimal as it is just a diff report but it will be able to detect changes on deep changes inside the validator.